flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/service/NativeFlinkService.java

```diff
@@ -201,8 +201,11 @@ public ScalingResult scale(FlinkResourceContext<?> ctx, Configuration deployConf
 
                 if (overrideStr != null) {
                     // We have an override for the vertex
-                    int p = Integer.parseInt(overrideStr);
-                    var newParallelism = new JobVertexResourceRequirements.Parallelism(p, p);
+                    int upperBound = Integer.parseInt(overrideStr);
+                    // We keep the current parallelism lower bound unless we went below it
+                    int lowerBound = Math.min(upperBound, parallelism.getLowerBound());
```

Comment:
Can we add a small comment why we do this?

Reply:
IIUC, Adaptive Scheduler will run the job without pause if we don't change the lowerBound.

Adaptive Scheduler can restart the job after the resource is ready.

Reply:
I will add some comments and do some extensive local testing before merging this